src/components/progress-circle/progress-circle.ts

```diff
@@ -41,48 +84,100 @@ export class MdProgressCircle {
+  }
 
 
   /**
-   * Gets the current stroke dash offset to represent the progress circle.
-   *
-   * The stroke dash offset specifies the distance between dashes in the circle's stroke.
-   * Setting the offset to a percentage of the total circumference of the circle, fills this
-   * percentage of the overall circumference of the circle.
+   * Animates the rendered circle.
    */
-  strokeDashOffset() {
-    // To determine how far the offset should be, we multiple the current percentage by the
-    // total circumference.
+  private _renderCircle(animateFrom: number, animateTo: number,
+                        ease: (currentTime: number, startValue: number,
+                               changeInValue: number, duration: number) => number,
+                        duration: number, rotation: number) {
+    let id = ++this._lastAnimationId;
+    let startTime = now();
+    let changeInValue = animateTo - animateFrom;
 
-    // The total circumference is calculated based on the radius we use, 45.
-    // PI * 2 * 45
-    return 251.3274 * (100 - this._value) / 100;
-  }
+    // No need to animate it if the values are the same
+    if (animateTo === animateFrom) {
+      this.currentPath = getSvgArc(animateTo, rotation);
+    } else {
+      let animation = (currentTime: number) => {
+        let elapsedTime = Math.max(
+          0, Math.min((currentTime || now()) - startTime, duration));
 
+        this.currentPath = getSvgArc(
+          ease(elapsedTime, animateFrom, changeInValue, duration),
+          rotation
+        );
 
-  /** Gets the progress value, returning the clamped value. */
-  get value() {
-    return this._value;
+        // Prevent overlapping animations by checking if a new animation has been called for and
+        // if the animation has lasted long than the animation duration.
+        if (id === this._lastAnimationId && elapsedTime < duration && pageVisible()) {
+          requestAnimationFrame(animation);
+        }
+      };
+      requestAnimationFrame(animation);
+    }
   }
 
 
-  /** Sets the progress value, clamping before setting the internal value. */
-  set value(v: number) {
-    if (v != null) {
-      this._value = MdProgressCircle.clamp(v);
+  /**
+   * Starts the indeterminate animation interval, if it is not already running.
+   */
+  private _startIndeterminateAnimation() {
+    var rotationStartPoint = 0;
+    var start = startIndeterminate;
+    var end =  endIndeterminate;
+    var duration = DURATION_INDETERMINATE;
+    var animate = () => {
+      if (pageVisible()) {
+        this._renderCircle(start, end, materialEase, duration, rotationStartPoint);
+        // Prevent rotation from reaching Number.MAX_SAFE_INTEGER.
+        rotationStartPoint = (rotationStartPoint + end) % 100;
+        let temp = start;
+        start = -end;
+        end = -temp;
+      }
+    };
+
+    if (!this._interdeterminateInterval) {
+      // Note that this interval isn't supposed to trigger a digest.
```

Comment:
We don't say the word "digest" any more 😉

When a `setInterval` function runs, it _will_ trigger a round of change detection unless you explicitly inject the `NgZone` and use that to `runOutsideAngular`.